Source/WebCore/html/parser/HTMLPreloadScanner.cpp

```diff
@@ -467,8 +469,18 @@ void TokenPreloadScanner::scan(const HTMLToken& token, Vector<std::unique_ptr<Pr
     case HTMLToken::Type::StartTag: {
         TagId tagId = tagIdFor(token.name());
         if (tagId == TagId::Template) {
-            ++m_templateCount;
-            return;
+            bool isDeclarativeShadowRoot = false;
+            static constexpr UChar shadowRootAsUChar[] = { 's', 'h', 'a', 'd', 'o', 'w', 'r', 'o', 'o', 't', 'm', 'o', 'd', 'e' };
+            const auto* shadowRootModeAttribute = findAttribute(token.attributes(), shadowRootAsUChar);
+            if (shadowRootModeAttribute) {
+                String shadowRootValue(shadowRootModeAttribute->value);
+                isDeclarativeShadowRoot = equalIgnoringASCIICase(shadowRootValue, "open"_s) || equalIgnoringASCIICase(shadowRootValue, "closed"_s);
+            }
+            // If this is a declarative shadow root <template shadowrootmode>
+            // element *and* we're not already inside a non-DSD <template> element,
+            // then we leave the template count at zero. Otherwise, increment it.
```

Comment:
I don't know what acronym "DSD" means here or in the bug title. Please expand it.

Reply:
Sure, I will update bug title plus comment and then will do `merge-queue` directly.

Reply:
Done. @anttijk - Changed in comment, commit message and title.